shared/src/main/scala/io/github/hamsters/HList.scala

```diff
@@ -257,11 +257,15 @@ case class HCons[T, U <: HList](head: T, tail: U) extends HList {
 
 object HList {
 
-  type ::[T,U <: HList] = HCons[T,U]
+  type ::[H,T <: HList] = HCons[H,T]
```

Comment:
If you change this you can change the U type  in the :: of the trait/implementations too?

Reply:
Ok , i will change it in all the class :)

Reply:
Should be H here I guess : https://github.com/scala-hamsters/hamsters/blob/6fb66ba359fa16a92a398ff19d19c85519605fe5/shared/src/main/scala/io/github/hamsters/HList.scala#L86